core/order.go

```diff
@@ -1028,9 +1063,10 @@ collectListings:
 	}
 
 	// Validate the each item in the order is for sale
+	knownListings := true
 	for _, listing := range contract.VendorListings {
 		if !n.IsItemForSale(listing) {
-			return errors.New("Contract contained item that is not for sale")
+			knownListings = false
```

Comment:
Again, this condition depends on one thing `n`. Instead, let's do `n.AnyListings()` or `n.HasListings()`.